### PR TITLE
Improve Assembly icon

### PR DESCRIPTION
**Pull request by ignamartinoli**
[what]
Adds consistence with other low level languages icons like C/C++.

[note]
Closes https://github.com/ryanoasis/nerd-fonts/issues/857.

@all-contributors please add @ignamartinoli code

Signed-off-by: igna_martinoli <ignamartinoli@protonmail.com>

#### Description

Closes [this](https://github.com/ryanoasis/nerd-fonts/issues/857) issue and improves the existent icon by making it more consistent with other low level languages icons.

#### Requirements / Checklist

- [X] Read the [Contributing Guidelines](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md)
- [X] Verified the license of any newly added font, glyph, or glyph set

#### What does this Pull Request (PR) do?

Improve the **Assembly** icon by making it consistent with other low level languages like **C**/**C++**

#### How should this be manually tested?

I really don't know.

#### Any background context you can provide?

Discussion [here](https://github.com/ryanoasis/nerd-fonts/issues/857)

#### What are the relevant tickets (if any)?

None.

#### Screenshots (if appropriate or helpful)

image

### CODE REVIEW

**Comment by Finii** [comment]
Thanks for the design an the PR.

I would not replace the existing 'asm', as it is part of Seti UI. We should keep that as we keep the other, simpler 'C' and 'C++' icons. All other nerd font's own svgs are now called `*_nf.svg`.
What is missing is an entry in the `icons.tsv`. (See `README.md` in `src/svgs/`.)

Do you want to amend the PR, or shall I?

**Review by Finii — CHANGES_REQUESTED** [review]
Please do not replace the Seti icon but rather add an additional one (with filename ending in `_nf.svg`. Also this needs a new entry in the icons 'database'.

**Comment by ignamartinoli** [comment]
I'm taking a shot at this. I plan on contributing more icons in the future so I want to be able to do it correctly :wink:

**Comment by Finii** [comment]
:+1: I hope it's not too many as the available slots are diminishing ;-)

**Comment by ignamartinoli** [comment]
Alright, I reverted the icon change and added the icon as `asm_nf.svg`.
I also changed `icons.tsv`, it looks like this

```
...
59		i_custom_asm				asm_nf.svg
# new
058		i_seti_kotlin
...
```

Reading through `src/svg/README.md`, should I run `generate-original-source.py` myself or is it going to run automatically at Push?

File: src/svgs/icons.tsv [hunk 1]
```diff
@@ -84,6 +84,7 @@
 56		i_custom_emacs				emacs_nf.svg
 57		i_custom_orgmode			orgmode_nf.svg
 58		i_custom_kotlin				kotlin_nf.svg
+59		i_custom_asm				asm_nf.svg
```
Comment (Finii) [inline src/svgs/icons.tsv:87]: offset 59 is already given to ... see below `059 i_seti_apple`.
You need to add it in the bottom.

I see, the switch from two-digits to three-digits can be confusing, as is the comment line in between. I ask myself how they come to be there :grimacing: Must have been forgotten.

**Comment by Finii** [comment]
> should I run `generate-original-source.py` myself

Not as part of the PR, but you can to see if your changes work.
That would have warned about the fact offset 59 is handed out twice... lets see

```
$ ./generate-original-source.py

[Nerd Fonts]  Glyph collection font generator 2.3.0-RC

Conflicting filename for 58677, ignoring apple.svg
Read glyph data successfully with 196 entries (19 aliases)
Generating original-source.otf with 177 glyphs
Generating GlyphInfo i_seti.sh
Finished
```

`optimize-original-source.sh` and `generate-original-source.py` are automatically run, the CI should have added commits to your PR automatically. Ah, because it is in your fork, it did not run, I assume. It will be run on pulling.

**Comment by Finii** [comment]
Icon itself looks good:

image

File: src/svgs/icons.tsv (hunk 1 above)
Comment (Finii) [reply src/svgs/icons.tsv]: Fixed

**Comment by Finii** [comment]
CI is broken on Github side.
The CI **needs** to run for `packsvgs.yml` :-(

Keeping to try, maybe it will be repaired in an hour or so.

**Comment by Finii** [comment]
Strange, why could the CI do this, while we could not...:

image

Is this a bug with `dry-run`? :thinking:

**Pull request merged**